test/e2e/runner/infra.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"context"
+
+	e2e "github.com/tendermint/tendermint/test/e2e/pkg"
+)
+
+// TestnetInfra provides an API for interacting with the infrastructure for an
+// entire testnet.
+type TestnetInfra interface {
+	// GenerateConfig generates any necessary configuration for the
+	// infrastructure provider during testnet setup.
+	GenerateConfig() error
+
+	// Stop attempts to stop the entire testnet.
+	Stop(ctx context.Context) error
+
+	// Pause attempts to pause the entire testnet.
+	Pause(ctx context.Context) error
+
+	// Unpause attempts to resume a paused testnet.
+	Unpause(ctx context.Context) error
+
+	// ShowLogs will print all logs for the whole testnet to stdout.
+	ShowLogs(ctx context.Context) error
```

Comment:
Need this return an error? Is there anything actionable the caller can do with an error here?
```suggestion
	// ShowLogs prints all logs for the testnet to stdout.
	ShowLogs(ctx context.Context) error
```

Reply:
Separately: I realize Haskell uses "show" for this concept, but I think "print" might be more descriptive. YMMV.

Reply:
This just seems like the convention used in the E2E tests. Returned errors are passed up through the stack to the Cobra CLI machinery and left to Cobra to deal with/display.